File: LinkedList/linkedlist.py
class node:
    def __init__(self,data):
        self.data=data
        self.next=None
    
class LinkedList:
    def __init__(self):
        self.head=None
    def listLength(self):
        count=0
        temp=self.head
        while(temp):
            count+=1
            temp=temp.next
        return count
    def listItems(self):
        temp=self.head
        if(self.head==None):
            print("Empty LinkedList\n")
            return
        while(temp):
            print(temp.data,)
            temp=temp.next
    def insert(self,data,position):
        count=1
        current=self.head
        temp=LinkedList()
        newNode=node(data)
        if(position==1):
            self.head=newNode
        else:
            while(current!=None and count<position):
                count+=1
                temp=current
                current=current.next    
        temp.next=newNode
        newNode.next=current
    def delete(self,position):
        count=1
        current=self.head
        temp=LinkedList()

        if(position==1):
            current=current.next
            del(self.head)
            self.head=current
        else:
            while(current!=None and (count<position)):
                temp=current
                current=current.next
            temp.next=current.next.next
            del(current)
    def findelement(self,data):
        temp=self.head
        count=1
        while(temp!=None):
            if(temp.data==data):
                print("element found at position :",count)
                return
            count+=1
            temp=temp.next
        
        print("element not found")
        return
    
if __name__=="__main__":
    llist = LinkedList()
    llist.insert(5,1)
    llist.insert(4,1)
    llist.insert(3,1)
    llist.insert(2,1)
    llist.insert(1,1)

    print(llist.listLength(),"\n")
    llist.listItems()

File: LinkedList/linkedlist.c
#include<stdio.h>
#include<stdlib.h>
struct ListNode
{
    int data;
    struct  ListNode * next;
}node;
struct ListNode *head=NULL;

int listLength(){
    struct ListNode * current=head;
    int count=0;
    while(current!=NULL){
        count++;
        current=current->next;
    }
    return count;
}
void insertAtPosition(int data,int position){
    struct ListNode * current=head;
    struct ListNode * temp=NULL;
    struct ListNode * newNode=(struct ListNode*)malloc(sizeof (struct ListNode));
    newNode->data=data;
    int count=1;
    if(position==1){
        newNode->next=current;
        head=newNode;
    }
    else{
        while((current!=NULL) && (count<position)){
            count++;
            temp=current;
            current=current->next;
        }
        temp->next=newNode;
        newNode->next=current;
    }
}
void listItems(){
    struct ListNode * current=head;
    
     if(head==NULL){
        printf("\n%s","No items in list");
        return;
    }
    printf("\n%s","List Items:");
    while(current!=NULL){
        printf(" %d",current->data);
        current=current->next;
    }
}
void deleteItem(int position){
    struct ListNode * current=head;
    struct ListNode* temp=NULL;
    int count=1;
    if(head==NULL){
        printf("\n%s","No items in list");
        return;
    }
    if(position==1){
        current=current->next;
        free(current);
    }
    else{
        while(current!=NULL && count<position){
            count++;
            temp=current;
            current=current->next;
        }
        temp->next=current->next;
        free(current);
    }
}
void DeleteList(){
    struct ListNode * auxilaryNode,*iterator;
    iterator=head;

    while(iterator){
        auxilaryNode=iterator->next;
        free(iterator);
        iterator=auxilaryNode;
    }
    head=NULL;

}
void findElementFromEnd(int position){
    struct ListNode * current=head;
    struct ListNode * temp;
    int num=listLength()-position;
    int k=0;
    while(current!=NULL && k<(num)){
        k++;
        current=current->next;

    }
    printf("%d",current->data);
}
void listReverse(){
    struct ListNode * current=head;\\
            temp=current;
            current=current->next;
        }
        temp->next=current->next;
        
        printf("%d",current->data);
        k--;
        
       
    }
    
}
int main(){
    insertAtPosition(1,1);
    insertAtPosition(2,2);
    insertAtPosition(3,3);
    insertAtPosition(4,4);
    insertAtPosition(5,5); 

    listReverse();
    listItems();

}